patches/server/0824-Forward-CraftEntity-in-teleport-command.patch

```diff
@@ -0,0 +1,25 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Sat, 4 Dec 2021 17:04:47 -0800
+Subject: [PATCH] Forward CraftEntity in teleport command
+
+
+diff --git a/src/main/java/net/minecraft/server/commands/TeleportCommand.java b/src/main/java/net/minecraft/server/commands/TeleportCommand.java
+index ced3b3ed227dacf1c4928ee2dfa4f390f13c8a8b..bdb800a2c5cc7bec2f4814310e2c3f23ee095f95 100644
+--- a/src/main/java/net/minecraft/server/commands/TeleportCommand.java
++++ b/src/main/java/net/minecraft/server/commands/TeleportCommand.java
+@@ -195,7 +195,14 @@ public class TeleportCommand {
+                     target.unRide();
+                     Entity entity1 = target;
+ 
++                    // Paper start - transfer bukkit entity
++                    org.bukkit.craftbukkit.entity.CraftEntity bukkitEntity = target.bukkitEntity;
+                     target = target.getType().create(world);
++                    if (bukkitEntity != null) {
++                        bukkitEntity.setHandle(target);
++                        target.bukkitEntity = bukkitEntity;
```

Comment:
Should we maybe move this logic into Entity#restoreFrom instead? It's only called in TeleportCommand and teleportTo, and seems to be a better place for this

Reply:
Yeah, and has the benefit of removing the need for the AT for the bukkitEntity field.

done.